src/libtrx/game/game_flow/sequencer.c

```diff
@@ -78,6 +82,101 @@ GF_COMMAND GF_InterpretSequence(
 
     GF_COMMAND gf_cmd = { .action = GF_EXIT_TO_TITLE };
 
+    const GF_LEVEL *const prev_level = GF_GetLevelBefore(level);
+
+    // before load
+    switch (seq_ctx) {
+    case GFSC_STORY:
+        break;
+
+    case GFSC_SAVED:
+        GF_InventoryModifier_Scan(level);
+        // reset current info to the defaults so that we do not do
+        // Item_GlobalReplace in the inventory initialization routines too early
+        Savegame_InitCurrentInfo();
+        break;
+
+    case GFSC_RESTART:
+        if (level == GF_GetGymLevel() || level == GF_GetFirstLevel()) {
+            Savegame_InitCurrentInfo();
+        } else {
+            Savegame_ResetCurrentInfo(level);
+            Savegame_CarryCurrentInfoToNextLevel(prev_level, level);
+            Savegame_ApplyLogicToCurrentInfo(level);
+        }
+        if (level->type == GFL_NORMAL || level->type == GFL_BONUS) {
+            GF_InventoryModifier_Scan(level);
+            GF_InventoryModifier_Apply(level, GF_INV_REGULAR);
+        }
+        break;
+
+    case GFSC_SELECT: {
+        const int16_t slot_num = Savegame_GetBoundSlot();
+        if (slot_num != -1) {
+            // select level feature
+            Savegame_InitCurrentInfo();
+            if (level->num > GF_GetFirstLevel()->num) {
+                Savegame_LoadOnlyResumeInfo(slot_num);
+                const GF_LEVEL *tmp_level = level;
+                while (tmp_level != nullptr) {
+                    Savegame_ResetCurrentInfo(tmp_level);
+                    tmp_level = GF_GetLevelAfter(tmp_level);
+                }
+                Savegame_CarryCurrentInfoToNextLevel(prev_level, level);
+                Savegame_ApplyLogicToCurrentInfo(level);
+                GF_InventoryModifier_Scan(level);
+                GF_InventoryModifier_Apply(level, GF_INV_REGULAR);
+            }
+        } else {
+            // console /play level feature
+            Savegame_InitCurrentInfo();
+            Inv_RemoveAllItems();
```

Comment:
We just need an `#include` for this, current compiler warning. My bad, should have sent the full diff.